app/controllers/reports/regions_controller.rb

```diff
@@ -8,14 +8,30 @@ class Reports::RegionsController < AdminController
   before_action :find_region, except: :index
   around_action :set_time_zone
 
-  def index
-    authorize(:dashboard, :show?)
+  skip_after_action :verify_authorized, if: -> { Flipper.enabled?(:new_permissions_system_aug_2020, current_admin) }
+  after_action :verify_authorization_attempted, if: -> { Flipper.enabled?(:new_permissions_system_aug_2020, current_admin) }
 
-    @organizations = policy_scope([:cohort_report, Organization]).order(:name)
+  def index
+    if Flipper.enabled?(:new_permissions_system_aug_2020, current_admin)
+      authorize1 { current_admin.accessible_facilities(:view_reports).any? }
+      @organizations = current_admin.accessible_facilities(:view_reports)
+        .includes(facility_group: :organization)
+        .flat_map(&:organization)
+        .uniq
+        .compact
```

Comment:
Which scope here introduces the dups and nils? Should we move these to inside `accessible_facilities` if that's the source? Should we move this to `accessible_organizations` and use that instead?

Reply:
`@organizations` is used to show links for organizations where the current_admin has access to any facilities under them. We need to `.uniq.compact` this so we only show one link for each organization.

We can't use `accessible_organizations` for this because in the new permissions system, having access to a single facility under an organization isn't the same as having access to the whole organization. This means `accessible_organizations` only returns organizations for power users and organization managers.

This worked in the old permissions system because the [:cohort_report, Organization] policy would traverse the tree in both directions and include organizations in the scope even if the user only had access to a single facility under them.

Reply:
👍  Does it make sense to add convenience methods for getting such orgs?